stdlib/2and3/socket.pyi

```diff
@@ -559,6 +560,28 @@ class socket:
     def setsockopt(self, level: int, optname: int, value: Union[int, bytes]) -> None: ...
     def shutdown(self, how: int) -> None: ...
 
+    if sys.version_info >= (3, 3):
+        @overload
+        def recvmsg(self, bufsize: int) -> Tuple[bytes, List[_CMSG], int, Any]: ...
```

Comment:
Why do these need overloads? Can't you just use default values?

Reply:
Using default values would mean that type checking wouldn't complain if one did this:

```python
sock.recvmsg(100, flags=socket.MSG_WAITALL)
```

But at run-time the following exception is raised:

```
TypeError: recvmsg() takes no keyword arguments
```

The intention was that overloading creates an error during static type checking so one doesn't have to wait for it to fail at run-time. But now that I'm testing it, the error produced by mypy is unhelpful and one can still do this:

```python
sock.recvmsg(100, 100, flags=5)
```

In which case mypy doesn't complain and the run-time error still occurs. :-(

According to the MyPy cheat sheet, I could append double underscores (__) to the parameters tomake them position-only arguments, but the mypy error is still unhelpful:

```
error: No overload variant of "recvmsg" of "socket" matches argument types [builtins.int, builtins.int, builtins.int]
```

Is there a better way? Or am I overthinking this one.

Reply:
What if you don't use overloads, but prefix the arguments with double underscores, like `def recvmsg(size: int, __flags: int = ...) -> SomeType: ...`?

In general, we've mostly not worried about positional-only arguments to builtins.

Reply:
Brilliant! Works like a charm. And the error message is meaningful too.